### PR TITLE
Fix file input focus traversal

### LINKED ISSUE #6372: FileInput - tab order and styling
When navigating with the keyboard the tab order of FileInput is unexpected. First our focus is on the 'browse' button hitting tab again moves our focus to the main portion of the FileInput. This can be seen in the video below. It feels strange for focus to move right to left across the screen. Also the main portion of the FileInput gives us context for what the input field is so it should come first in the tab order before the browse button.

Additionally when not using a screenreader the main portion of the FileInput doesn't receive a focusIndicator. When navigating with the keyboard we should visually show a focus indicator on this section.

This issue is reproducable on the FileInput storybook stories: https://storybook.grommet.io/?path=/story/input-fileinput-max-size--max-size

Video of the issue with screenreader output:

https://user-images.githubusercontent.com/54560994/193368791-0fd2f60f-b1dd-4e55-9c74-4228dc80436a.mp4

### PR DESCRIPTION
**Pull request by miqh**
Hello there,

These changes are intended to address #6372 following some discussion over there.

---

#### What does this PR do?
It revises the keyboard navigation focus for the file input component. There's only one focus stop and it corresponds to the actual HTML file input element. Refer to the screen captures I've attached below.

#### Where should the reviewer start?
All of the changes are in the file input component implementation.

#### What testing has been done on this PR?
Manual verification through the file input stories available on Storybook.

#### How should this be manually tested?
By using keyboard navigation on any of the available file input stories.

#### Do Jest tests follow these best practices?

I haven't included any additional tests to cover the adjusted traversal behaviour, but happy to do so if necessary.

I looked around but didn't spot (or overlooked) any other components which had tests specifically for testing keyboard navigation accessibility.

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
None.

#### What are the relevant issues?
Refer to #6372

#### Do the grommet docs need to be updated?
Unsure. Maintainers to decide.

#### Should this PR be mentioned in the release notes?
Unsure. Maintainers to decide.

#### Is this change backwards compatible or is it a breaking change?
Backward compatibility should not be affected.

---

Closes #6372

### CODE REVIEW

File: src/js/components/FileInput/FileInput.js [hunk 1]
```diff
@@ -306,6 +352,7 @@ const FileInput = forwardRef(
                   >
                     {theme.fileInput.button ? (
                       <Button
+                        tabIndex={-1}
```
Comment (jcfilben) [inline src/js/components/FileInput/FileInput.js:355]: Can we add a comment to the places we set the `tabIndex={-1}`?
```suggestion
                        // Button is redundant for keyboard users
                        tabIndex={-1}
```

Comment (miqh) [reply src/js/components/FileInput/FileInput.js:356]: @jcfilben, we sure can. I've amended my commit with the extra commentary.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by jcfilben — COMMENTED** [review]
One more thing, can we only show the focus style when we are using the keyboard? We do this on other components

**Comment by miqh** [comment]
@jcfilben, sure.

I've brought in the `useKeyboard` hook so that the file input element focus is made conditional on that.

The `Select` implementation is much meatier than `FileInput`, so if this isn't the behaviour you had in mind please point me in the right direction. 🙂

As an aside, thought I'd also mention that the `Anchor` component appears to trigger a focus outline on `mousedown`. This behaviour doesn't look immediately customisable from the `props` available.

**Comment by jcfilben** [comment]
> @jcfilben, sure.
>
> I've brought in the useKeyboard hook so that the file input element focus is made conditional on that.
>
> The Select implementation is much meatier than FileInput, so if this isn't the behaviour you had in mind please point me in > the right direction. 🙂

Nice work, this looks good to me

> As an aside, thought I'd also mention that the Anchor component appears to trigger a focus outline on mousedown. This behaviour doesn't look immediately customisable from the props available.

Yeah I think there are some components that still aren't handling focus in the ideal way. My inclination is that Anchor shouldn't get a focusIndicator unless using the keyboard. I realize we do have a bit of inconsistency across the library with this. I'll bring this up with some other members of the grommet team but feel free to share any more thoughts if you notice other components that aren't handling focus the way you would expect them to.

**Review by taysea — APPROVED** [review]
Nice work!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**